src/main/java/com/demo/classroom/Service/CourseService.java

```diff
@@ -0,0 +1,65 @@
+package com.demo.classroom.Service;
+
+import java.util.List;
+import java.util.ArrayList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.demo.classroom.DTO.ApiResponse;
+import com.demo.classroom.DTO.BookDTO;
+import com.demo.classroom.DTO.CourseDTO;
+import com.demo.classroom.Entity.Book;
+import com.demo.classroom.Entity.Course;
+import com.demo.classroom.Entity.Teacher;
+import com.demo.classroom.Repository.CourseRepository;
+import com.demo.classroom.Repository.TeacherRepository;
+import com.demo.classroom.Security.Service.JwtService;
+import com.demo.classroom.Utility.Constants;
+
+import java.util.Optional;
+
+@Service
+public class CourseService {
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Autowired
+    private TeacherRepository teacherRepository;
+
+    @Autowired
+    private JwtService jwtService; 
+
+    public ApiResponse<Void> createCourse(CourseDTO courseDTO, String token) {
+        Long teacherId = jwtService.extractUserId(token);  
+        Optional<Teacher> teacher = teacherRepository.findByUserId(teacherId);
+       
+        if (teacher.isEmpty()) {
+           return createApiResponse(true, Constants.USER_NOT_FOUND.getMessage());
+        }
```

Comment:
The message returned (Constants.USER_NOT_FOUND.getMessage()) could be adjusted to reflect a failure status rather than success (e.g., changing the first parameter to false).
This would help clarify that the operation did not complete successfully due to a missing teacher.

Reply:
solved